src/routes/extras/containers/ExtrasContainer.js

```diff
@@ -70,31 +70,43 @@ class ExtrasContainer extends React.Component<ExtrasPropsType, ExtrasStateType>
     }
   }
 
-  async loadExtras () {
+  loadExtras = async () => {
     const { city, language } = this.props
-    const payload: Payload<Array<ExtraModel>> = await (createExtrasEndpoint(baseUrl).request({ city, language }))
-
-    if (payload.error) {
-      this.setState(() => ({ error: payload.error, extras: null }))
-    } else {
-      this.setState(() => ({ error: null, extras: payload.data }))
+    this.setState({ error: null, extras: null })
+
+    try {
+      const payload: Payload<Array<ExtraModel>> = await (createExtrasEndpoint(baseUrl).request({ city, language }))
+
+      if (payload.error) {
+        this.setState({ error: payload.error, extras: null })
+      } else {
+        this.setState({ error: null, extras: payload.data })
+      }
+    } catch (e) {
+      this.setState({ error: e, extras: null })
     }
   }
 
   render () {
     const { theme, t, cities, navigation, city, language } = this.props
     const { extras, error } = this.state
 
-    if (error || !cities) {
-      return <Failure error={error} theme={theme} t={t} />
+    if (error) {
+      return <ScrollView refreshControl={<RefreshControl onRefresh={this.loadExtras} refreshing={false} />}
+                         contentContainerStyle={{ flexGrow: 1 }}>
+        <FailureContainer error={error} tryAgain={this.loadExtras} />
+      </ScrollView>
     }
 
-    if (!extras) {
-      return <ActivityIndicator size='large' color='#0000ff' />
+    if (!extras || !cities) {
+      return <ScrollView refreshControl={<RefreshControl refreshing />} contentContainerStyle={{ flexGrow: 1 }} />
     }
 
-    return <Extras extras={extras} navigateToExtra={this.navigateToExtra} theme={theme} t={t} cities={cities}
-                   navigation={navigation} cityCode={city} language={language} />
+    return <ScrollView refreshControl={<RefreshControl onRefresh={this.loadExtras} refreshing={false} />}
+                       contentContainerStyle={{ flexGrow: 1 }}>
+      <Extras extras={extras} navigateToExtra={this.navigateToExtra} theme={theme} t={t} cities={cities}
```

Comment:
Is there no ScrollView in the Extras component?

Reply:
Not anymore :stuck_out_tongue: